src/main/java/com/lorepo/icplayer/client/module/ordering/OrderingPrintable.java

```diff
@@ -21,63 +21,160 @@ public OrderingPrintable(OrderingModule model, PrintableController controller) {
+	 *                           String type. Numbers represent indexes of items. The numbers are arranged in order.
+	 *                           Numbers must be separated by a comma.
+	 * @return                indexes items in order from the state
+	 */
+	private String[] getItemsIndexesInOrderFromState(HashMap<String, String> printableState) {
+		String orderState = printableState.get("order");
+		return orderState.split(",");
+	}
+
+	/**
+	 * createPrintableItem  Returns a printable representation of the item.
+	 * @param  item         object whose representation is created
+	 * @param  index        position of object. index == 0 means index will not be displayed
+	 * @param  isVertical   if the object must be represented vertically
+	 * @param  checkAnswers if item will be represented in the checkAnswers state
+	 * @return              printable representation of the item
+	 */
+	private String createPrintableItem(OrderingItem item, int index, boolean isVertical, boolean checkAnswers) {
+		String displayMode = isVertical? "block" : "inline-block";
+		String result = "<div style=\"display:" + displayMode + "\">";
+		result += "<div class=\"item-wrapper\">";
+		result += "<table><tr><td>";
+		result += createPrintableIndexBox(item, index, checkAnswers);
+		result += "</td><td>";
+		result += item.getText();
+		result += "</td></tr></table></div></div>";
+		return result;
+	}
+
+	/**
+	 * createPrintableIndexBox Returns a printable representation of the item's box with index.
+	 * @param  item         object for which the box representation is created
+	 * @param  index        position of object. index == 0 means index will not be displayed
+	 * @param  checkAnswers if true, add a sign representing whether the indexes match
+	 * @return              printable representation of the item's box
+	 */
+	private String createPrintableIndexBox(OrderingItem item, int index, boolean checkAnswers) {
+		String box = "";
+		if (checkAnswers) {
+			if (item.isCorrect(index))
+				box = "<div class=\"number-box-correct\">";
+			else
+				box = "<div class=\"number-box-wrong\">";
+		} else {
+			box = "<div class=\"number-box\">";
+		}
+		if (index != 0) {
+			box += Integer.toString(index);
 		}
-		
+		box += "</div>";
+		return box;
+	}
+
+	private void sortItemsRepresentations(String[] parsedItems, List<String> unorderedItems){
+		for(int index = 0; index < unorderedItems.size(); index += 1) {
+			Collections.swap(unorderedItems, index, index + nextInt(unorderedItems.size() - index));
+		}
+
 		for (int i = 0; i < parsedItems.length; i++) {
 			if (parsedItems[i] == null) {
 				parsedItems[i] = unorderedItems.get(0);
 				unorderedItems.remove(0);
 			}
 		}
-		
-		for (int i = 0; i < parsedItems.length; i++) {
-			result += parsedItems[i];
+	}
+
+	private String combineItemsRepresentations(String[] printableItems, boolean checkAnswers, boolean isAllCorrect) {

```

Comment:
Starałbym się unikać przekazywania booleanów jako flag kontrolujących zachowanie funkcji - jako że wywołanie
combineItemsRepresentations(printable, false, false), a
combineItemsRepresentations(printable, false, true) bardzo niewiele mówi.

Tutaj łatwo to zmienić na:
1. Wygenerowanie tego co jest w "items-wrapper"
2. Wygenerowanie znacznika "all-ok"
3. Wsadzenie wygenerowanych treści w diva "printable_ic_ordering"

Takie problemy opisane są np. tutaj:
https://martinfowler.com/bliki/FlagArgument.html i powiązane
https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3

Reply:
Przyjrzę się

Reply:
Mam nadzieję że teraz wygląda lepiej.